### ISSUE TITLE
Memory not freed after rocksdb_get

### ISSUE THREAD

**Issue by Sander-V-D** [maiha/rocksdb.cr#4]
Hey! I noticed that the memory usage of programs using this shard increases over time with the number of rocksdb_get calls.

Here is a minimal example: Initialise the database, and input some values. Running this program (on Ubuntu 18.04) results in increased memory usage until the program is killed.

```
require "rocksdb"
db = RocksDB::DB.new("path/to/rocksdb")
keys = db.keys
loop do
  keys.each do |k|
    puts db.get(k)
  end
  GC.collect
end
```

It seems Crystal can't free the memory that was returned in the db.get call. See also [here](https://forum.crystal-lang.org/t/crystal-shard-rocksdb-memory-leak/3466), where asterite suggests hooking into the Crystal GC to avoid having to worry about it further.

**Comment by maiha** [comment]
@Sander-V-D  Nice catch!
Please try the following modified branch. If it's resolved, we'd like to release it.

```yaml
dependencies:
  rocksdb:
    github: maiha/rocksdb.cr
    branch: fix-memory-leak-db-get
```

**Comment by Sander-V-D** [comment]
I can confirm that this fixes the issue for my program! Thanks for the quick fix!

**Comment by carlhoerberg** [comment]
Don't you have the same problem for `get?(key : Bytes)` then: https://github.com/maiha/rocksdb.cr/blob/1706f9774a3143255cbc94c35cbf644c7a78fe0e/src/rocksdb/commands/basic.cr#L4-L7

the pointer is never freed. either document that the user must `rocksdb_free` the pointer, or copy the content to a new GC managed `Bytes` array like so:

```crystal
@len.value == 0 ? nil : Bytes.new(@len.value).tap &.copy_from(ptr, @len.value)
```

**Comment by maiha** [comment]
@carlhoerberg 
Right! Released 1.1.0. Thank you 👍 

rocksdb-1.0.1
```console
$ crystal samples/test_memory_leak_bytes.cr 
15.4 MB
26.5 MB
36.3 MB
46.1 MB
```

rocksdb-1.1.0
```console
$ crystal samples/test_memory_leak_bytes.cr
15.5 MB
16.6 MB
16.6 MB
16.6 MB
```
